### PR TITLE
Bump version to 0.6.0

### PR DESCRIPTION
**Pull request by torkleyy**

### CODE REVIEW

**Comment by kvark** [comment]
Is this just for `base64`? I'd rather not have a dedicated release then, for now, until we have more breaking changes accumulated.

**Comment by kvark** [comment]
@CAD97 if you are interested in this, we basically need a short overview of all changes that went in since the last release. Admittedly, we are missing a tag/branch for latest 0.5 release, this is no good and need to be fixed as well. Once this little investigation is done, we can both create a branch for 0.5, and add a CHANGELOG listing at least stuff changed between 0.5 and 0.6, following up with the sections for earlier revisions later.

Also, all the pending issues that would require API changes need to be triaged for the prospect of including into this breaking release.

If you (or anyone) are willing to help with this, it would be great!

**Comment by CAD97** [comment]
Checking the source downloaded from crates-io, the published commit seems to be 6e9f7dd3080bdedb88ad89700da1aad499cb303f for 0.5.1. (Methodology: 0.5.1 was published on Apr 5. Above commit is a bors commit on Apr 5. The code landed by that merge is present. The code from the following commit, d25ea2055e20170d663fb5f5d9632ee6f4202a9a, appears to not be present.)

That means all of the following PRs have landed but not yet been published:

- #157 (Fixed #100, #155)
- #160 
- #164
- #163 (Fixed #159)
- #161 (Fixed #158)
- #166 (Addressed #123)
- #170 (Fixed #117)
- #171 (Fixed #167)
- #173 
- #172 (Fixed #129)
- #187 
- #182 
- #188 
- #193 
- #195 
- #197 
- #204 
- #208 
- #219 

If we're talking about fitting in more breaking changes, it would be great if ron could conform to [the serde data format conventions](https://serde.rs/conventions.html) for API surface, which is basically just

```rust
mod de;
mod error;
mod ser;

pub use de::{from_str, Deserializer};
pub use error::{Error, Result};
pub use ser::{to_string, Serializer};
```

I'll look into what's required for that.

**Comment by kvark** [comment]
Thank you! At the minimum, we still need to triage all the open issues to see, which of them require breaking changes and can be done.

**Comment by CAD97** [comment]
<details><summary>Triage</summary>

- #47: breaking
- #85: external
- #91: breaking, but honestly I don't think we want a date primitive
- #115: additive
- #122: breaking
- #123: tests
- #140: additive (but with a rename suggested)
- #143: additive
- #144: additive
- #152: external? fixed?
- #174: breaking (bugfix)
- #175: internal?
- #176: meta
- #178: probably internal
- #180: help
- #185: meta, external(?)
- #189: probably breaking
- #191: external
- #192: help
- #200: additive
- #201: help
- #202: additive
- #203: additive (probably)
- #207: additive (probably)
- #216: external, additive
- #217: additive (maybe impossible?)
</details>

I'll be honest: poking around in ron's code, I'm starting to feel that we could (should?) just start over, especially because of #175. Specifically, migrate `ron` to be just about parsing the data format, and create `serde_ron` to handle the serde part (the way that `serde_yaml` uses `yaml-rust`).

I've already [implemented a "better" parser](https://github.com/ron-rs/ron/issues/178#issuecomment-538812411) once, and would be happy to do the implementation work to get said split design up to parity with this version.

TL;DR: I think go ahead and publish 0.6 as-is, and start work on 0.7 that's pretty much a fresh start. And apparently I'm volunteering to do the initial impl work for the 0.7 rewrite.

**Comment by kvark** [comment]
Thank you for classifying the issues! Looks like we can fix #174  and indeed go ahead with 0.6.
@torkleyy any feelings about this?

>  And apparently I'm volunteering to do the initial impl work for the 0.7 rewrite.

That's a big task, but it looks like you are capable of accomplishing it. This is much appreciated!

**Comment by kvark** [comment]
Closing in favour of #234
